### PR TITLE
Add `name` filter to `Repository.get_artifacts()`

### PR DESCRIPTION
**Pull request by trim21**
https://docs.github.com/en/rest/actions/artifacts?apiVersion=2022-11-28#list-artifacts-for-a-repository

### CODE REVIEW

**Review by EnricoMi — APPROVED** [review]
LGTM!

**Comment by EnricoMi** [comment]
@trim21 can you add a test for this please?

**Comment by trim21** [comment]
> @trim21 can you add a test for this please?

done

File: tests/Artifact.py [hunk 1]
```diff
@@ -19,6 +19,14 @@ def testGetArtifactsFromWorkflow(self):
             repr(artifact), 'Artifact(name="vscode-codeql-extension", id=10495898)'
         )
 
+    def testGetArtifactsFromWorkflowWithName(self):
+        artifact = self.repo.get_artifacts(name="vscode-codeql-extension")[0]
```
Comment (EnricoMi) [inline tests/Artifact.py:23]: This does not test there is exactly one artifact, it only tests the first returned artifact is as expected.

File: tests/Artifact.py [hunk 2]
```diff
@@ -19,6 +19,14 @@ def testGetArtifactsFromWorkflow(self):
             repr(artifact), 'Artifact(name="vscode-codeql-extension", id=10495898)'
         )
 
+    def testGetArtifactsFromWorkflowWithName(self):
```
Comment (EnricoMi) [inline tests/Artifact.py:22]: Your test name suggests this gets artifacts for a workflow. Given the name, we would expect only a single artifact. That is the reason for my confusion. Test should be renamed:

```suggestion
    def testGetArtifactsFromRepoWithName(self):
```

Comment (EnricoMi) [reply tests/Artifact.py:22]: Would you consider to add the name filter to `Repository.get_workflow_run(...).get_artifacts` as well?

File: tests/Artifact.py (hunk 1 above)
Comment (EnricoMi) [reply tests/Artifact.py:23]: We could always move to a different repo.

Comment (EnricoMi) [reply tests/Artifact.py:23]: Assert that we expect multiple artifacts here:
```suggestion
        artifacts = self.repo.get_artifacts(name="vscode-codeql-extension")
        self.assertEqual(artifacts.totalCount, 5420)
        artifact = artifacts[0]
```

**Review by EnricoMi — CHANGES_REQUESTED** [review]

File: tests/Artifact.py (hunk 2 above)
Comment (trim21) [reply tests/Artifact.py]: Maybe later in another PR, not in this one.

Comment (trim21) [reply tests/Artifact.py]: after pyi are merged back to py

**Review by EnricoMi — APPROVED** [review]
LGTM!

**Pull request merged**